File: BackendCitasMedicas/src/main/java/com/example/BackendCitasMedicas/Repository/MedicoRepository.java
package com.example.BackendCitasMedicas.Repository;

import com.example.BackendCitasMedicas.Model.Medico;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface MedicoRepository extends JpaRepository<Medico, Long> {
}

File: FrontendCitasMedicas/app/src/main/java/com/example/frontcitasmedicas/Repository/MedicoRepository.kt
package com.example.frontcitasmedicas.Repository

import com.example.frontcitasmedicas.Model.Medico
import com.example.frontcitasmedicas.Service.RetrofitClient

class MedicoRepository {

    private val api = RetrofitClient.medicoService

    suspend fun getMedicos() = api.getAllMedicos()

    suspend fun getMedico(id: Long) = api.getMedicoById(id)

    suspend fun addMedico(medico: Medico) = api.addMedico(medico)

    suspend fun updateMedico(medico: Medico) = api.updateMedico(medico)

    suspend fun deleteMedico(id: Long) = api.deleteMedico(id)
}
